sdk/src/client/node_api/indexer/mod.rs

```diff
@@ -28,8 +28,11 @@ impl ClientInner {
             items: Vec::new(),
         };
 
-        // Return early with only a single page if a `QueryParameter::Cursor(_)` is provided.
-        let return_early = query_parameters.contains(QueryParameter::Cursor(String::new()).kind());
+        let query_string = query_parameters.to_query_string();
+        // Return early with only a single page if an empty string is provided as cursor.
+        let return_early = query_string
+            .map(|s| s.contains("cursor=&") || s.ends_with("cursor="))
```

Comment:
Wouldn't `s.contains("cursor=")` be enough ?

Reply:
Or maybe you just add `has_cursor` method idk

Reply:
regex?

Reply:
It's probably better that the check happens before the serialisation though

Reply:
If there is `cursor`, but with an actual specified value, then we don't want to return early, that's why just checking for `cursor`  is not enough